sagemaker-pyspark-sdk/src/sagemaker_pyspark/wrapper.py

```diff
@@ -157,3 +160,27 @@ def _to_java(self):
 
     def _from_java(self):
         raise NotImplementedError()
+
+
+# wrap scala.collection.immutable.List
+class ScalaList(SageMakerJavaWrapper):
+
+    _wrapped_class = "scala.collection.immutable.List"
+
+    def __init__(self, p_list):
+        self.p_list = p_list
+
+    def _to_java(self):
+        # Since py4j cannot deal with scala list directly
+        # we convert to scala listmap as an intermediate step
+        s_list = self._new_java_obj("scala.collection.immutable.ListMap")
+        key_list = range(len(self.p_list))
+        for elem, key in zip(self.p_list, key_list):
```

Comment:
I really think it would be cleaner to keep


for key, elem in enumerate(self.P_list):

Reply:
Fixed! Thanks